subprojects/core/src/main/java/org/gradle/api/Task.java

```diff
@@ -562,6 +563,13 @@ public String determineName(Task c) {
     TaskOutputs getOutputs();
 
     /**
+     * <p>Returns the files that this task destroys (removes).</p>
+     * @return The files that this task destroys.  Never returns null.
+     */
+    @Internal
+    TaskDestroys getDestroys();
```

Comment:
I settled on "destroyables".  I'm not sure this is great, but per our conversation it seems like the least worst option.  I've updated classes/methods/docs accordingly.

`@Incubating` and `@since` also added.